contracts/v6/OraclePrice.sol

```diff
@@ -0,0 +1,111 @@
+import "./interface/EACAggregatorProxy.sol";
+import "./library/SafeMath.sol";
+import "./common/Ownable.sol";
+
+contract OraclePrice is Ownable, KeeperCompatibleInterface {
+    using SafeMath for uint256;
+
+    struct OracleInfo {
+        address oracleAddress;
+        bool status;
+        bytes32 symbolHash;
+        int256 lastPrice; 
+    }
+    
+    OracleInfo[] public oracleInfo;
+    address public policyContract;
+    uint256 public deviationThreshold;
+    uint public immutable interval;
+    uint public lastTimeStamp;
+    uint public counter;
+    
+    constructor(OracleInfo[] memory _oracles, address _policyContract, uint256 _deviationThreshold, uint _updateInterval) public {
+        policyContract = _policyContract;
+        deviationThreshold = _deviationThreshold;
+        
+        for(uint256 i=0; i<_oracles.length; i++) {
+            OracleInfo memory oracles = _oracles[i];
+            oracleInfo.push(OracleInfo({
+                oracleAddress: oracles.oracleAddress,
+                status: oracles.status,
+                symbolHash: oracles.symbolHash,
+                lastPrice: 0
+            }));
+        }
+
+        interval = _updateInterval;
+        lastTimeStamp = block.timestamp;
+        counter = 0;
+    }
+    
+    function getOraclePriceInUsd(uint256 _oracleId) public view returns (int256) {
+        OracleInfo storage oracles = oracleInfo[_oracleId];
+        int256 latestPrice = EACAggregatorProxy(oracles.oracleAddress).latestAnswer();
+        return latestPrice;
+    }
+
+    function checkUpkeep(bytes calldata checkData) external override returns (bool upkeepNeeded, bytes memory performData) {
+        upkeepNeeded = (block.timestamp - lastTimeStamp) > interval;
+    }
+
+    function performUpkeep(bytes calldata performData) external override {
+        bool upkeepNeeded = (block.timestamp - lastTimeStamp) > interval;
+        require(upkeepNeeded == true, "Can Not call this method at this time");
+
+        lastTimeStamp = block.timestamp;
+        counter = counter + 1;
+
+        updateTargetPrice();
+    }
+    
+    function updateTargetPrice() internal {
+        uint256 length = oracleInfo.length;
+        for(uint256 i=0; i<length; i++) {
+            OracleInfo storage oracles = oracleInfo[i];
+            if(oracles.status == true) {
+                int256 latestPrice = EACAggregatorProxy(oracles.oracleAddress).latestAnswer();
+                uint8 decimals = EACAggregatorProxy(oracles.oracleAddress).decimals();
+                uint256 restDec = SafeMath.sub(18, uint256(decimals));
+                latestPrice = int256(SafeMath.mul(uint256(latestPrice), 10**restDec));
+                oracles.lastPrice = latestPrice;
+            }
+        }
+        uint256 targetRate = IOmsPolicy(policyContract).targetPrice();
+        uint256 rate  = IOmsPolicy(policyContract).calculateReferenceRate();
+        bool status = withinDeviationThreshold(rate, targetRate);
```

Comment:
It make it clearer, perhaps this should be called shouldUpdatePrice instead of simply status?

Reply:
Okay updating